hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerProtocolServerSideTranslatorPB.java

```diff
@@ -218,6 +219,7 @@ private OMResponse internalProcessRequest(OMRequest request) throws
         this.lastRequestToSubmit = requestToSubmit;
       } catch (IOException ex) {
         if (omClientRequest != null) {
+          OMAuditLogger.log(omClientRequest.getAuditBuilder());
```

Comment:
Should we use the other log method that takes a throwable instance so as to show that the request failed in preExecute?

Reply:
For pre-execute, audit log message is build inside the pre-execute handler where its required for IOException. We are not logging all failure in pre-execute, like authorization and others. So kept similar behavior.

Reply:
For the preExecute case, markForAudit has not been called so the setLog would be set to true and this wouldn't write anything to audit log.
`if (builder.isLog.get()) {` this would be false always

Reply:
Based on offline discussion , We don't write audits for failures in preExecute in every case. Some cases where audits are written in preExecute phase:
1. OMCancelDelegationTokenRequest
2. OMGetDelegationTokenRequest
3. OMRenewDelegationTokenRequest
The same behaviour is retained with this patch too.